src/components/rule-builder/rule-builder.component.tsx

```diff
@@ -0,0 +1,1746 @@
+        isAllowFutureDate ? setIsAllowFutureDate(false) : setIsAllowFutureDate(true);
+      }
+    }, [checkIfDateValidatorExists, isAllowFutureDate, onSchemaChange, pageIndex, questionIndex, schema, sectionIndex]);
+
+    /**
+     * Handle the "Disallow Decimal Values" for a target form field.
+     * 1. Checks if the "Disallow Decimals" option exists for the question.
+     * 2. Hanlder function to "Disallow Decimals" option and updates the schema.
+     */
+    const checkIfDecimalValidatorExists = useCallback(() => {
+      return !!schema.pages[pageIndex].sections[sectionIndex].questions[questionIndex].questionOptions.disallowDecimals;
+    }, [pageIndex, questionIndex, schema.pages, sectionIndex]);
+
+    const handleDisallowDecimalValueChange = useCallback(() => {
+      const doesValidatorExist = checkIfDecimalValidatorExists();
+      const updatedSchema = { ...schema };
+      const questionOptions =
+        updatedSchema.pages[pageIndex].sections[sectionIndex].questions[questionIndex].questionOptions;
+      questionOptions.disallowDecimals = doesValidatorExist ? !questionOptions?.disallowDecimals : true;
+      setIsDisallowDecimals(questionOptions.disallowDecimals);
+      onSchemaChange(updatedSchema);
+    }, [checkIfDecimalValidatorExists, onSchemaChange, pageIndex, questionIndex, schema, sectionIndex]);
+
+    /**
+     * Handle the "Historical Expression" property for a target form field.
+     * 1. Checks if the "Historical Expression" property exists for the question.
+     * 2. Handler function to enable/disable "Historical Expression" property within the schema.
+     */
+    const checkIfHistoricalExpressionExists = useCallback(() => {
+      return !!schema?.pages[pageIndex]?.sections[sectionIndex].questions[questionIndex]?.historicalExpression;
+    }, [pageIndex, questionIndex, schema?.pages, sectionIndex]);
+
+    const handleHistoryChange = useCallback(() => {
+      const isHistoricalExpressionExist = checkIfHistoricalExpressionExists();
+      const updatedSchema = { ...schema };
+      const historicalExpressionSchema = `HD.getObject('prevEnc').getValue('${question.questionOptions.concept}')`;
+      if (isHistoricalExpressionExist) {
+        delete schema.pages[pageIndex].sections[sectionIndex].questions[questionIndex].historicalExpression;
+      } else {
+        schema.pages[pageIndex].sections[sectionIndex].questions[questionIndex].historicalExpression =
+          historicalExpressionSchema;
+      }
+      setIsHistoryEnable((p) => !p);
+      onSchemaChange(updatedSchema);
+    }, [
+      checkIfHistoricalExpressionExists,
+      onSchemaChange,
+      pageIndex,
+      question.questionOptions.concept,
+      questionIndex,
+      schema,
+      sectionIndex,
+    ]);
+
+    /**
+     * Build the schema for a "target" condition.
+     * Generates the condition expression based on the provided parameters.
+     */
+    const getSchemaCondition = (
+      condition: string,
```

Comment:
Shouldn't `condition` here have a stricter type, i.e., to correspond to the set of known conditions?